app/server/api/search_api.py

```diff
@@ -31,33 +32,110 @@ def get(self):
+                                        'recipient_first_name': recipient.first_name,
+                                        'recipient_last_name': recipient.last_name,
+                                        'recipient_email': recipient.email,
+                                        'recipient_date_account_created': recipient.created
+                                        }
+
+        # These lists are to validate the user input-- not using sort_types_to_database_types since credit_transfers and transfer_accounts have unique options 
+        user_sorting_options = ['first_name', 'last_name', 'email', 'date_account_created']
+        sender_sorting_options = list(map(lambda s: 'sender_'+s, user_sorting_options)) # sender_first_name, sender_last_name, etc...
+        recipient_sorting_options = list(map(lambda s: 'recipient_'+s, user_sorting_options)) # recipient_first_name, recipient_last_name, etc...
+        sorting_options = { 'transfer_accounts': [*user_sorting_options, 'rank', 'balance', 'status'] ,
+                            'credit_transfers':[*sender_sorting_options, *recipient_sorting_options, 'rank', 'amount', 'transfer_type', 'approval', 'date_transaction_created'] }
+        sort_by_arg = request.args.get('sort_by') or 'rank'
+        if sort_by_arg not in sorting_options[search_type]:
+            response_object = {
+                # Example output:
+                # "Invalid sort_by value 'pizza'. Please use one of the following: 'first_name', 'last_name', 'email', 'rank', 'balance', 'status', 'date_account_created'"
+                'message': 'Invalid sort_by value \'{}\'. Please use one of the following: {}'\
+                    .format(sort_by_arg, ', '.join('\'{}\''.format(a) for a in sorting_options[search_type])),
+            }
+            return make_response(jsonify(response_object)), 400
+        sort_by = sort_types_to_database_types[sort_by_arg]
+
+        # HANDLE PARAM : order
+        # Valid orders types are: `ASC` and `DESC`
+        # Default: DESC
+        order_arg = request.args.get('order') or 'DESC'
+        if order_arg not in ['ASC', 'DESC']:
+            response_object = {
+                'message': 'Invalid order value \'{}\'. Please use \'ASC\' or \'DESC\''.format(order_arg),
+            }
+            return make_response(jsonify(response_object)), 400
+        order = desc
+        if order_arg == 'ASC':
+            order = asc
+
         # Note: Using tsquery wildcards here. Good docs of them here:
         # https://www.postgresql.org/docs/current/datatype-textsearch.html#DATATYPE-TSQUERY
         # 'Fran deRoo' -> 'Fran:* | deRoo:*'
         # Matches strings like "Francine deRoos"
         # Will also match "Michiel deRoos" because of the or clause, but this will be ranked lower
-        search_string = re.sub('\s+',' ',search_string)
         search_terms = search_string.strip().split(' ')
         tsquery = ':* | '.join(search_terms)+':*'
```

Comment:
Removing `search_string = re.sub('\s+',' ',search_string)` means that when I send a query 
 such as `http://0.0.0.0:9000/api/v1/search/?search_string=tristan%20%20%2B61401391419&search_type=credit_transfers&org=2` with two paces between a search string it will raise a 500 error -> ```sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) syntax error in tsquery: "tristan:* | :* | +61401391419:*"```

Reply:
Was there a reason you removed the strip?

Reply:
This (and a couple others) are just from me accidentally rolling back a merge conflict fix...